pkg/server/cloud/aws/instances.go

```diff
@@ -441,6 +466,41 @@ func (e *AwsEC2) retrieveOrAllocateHost(node *api.Node) (*string, error) {
 	return allocateOutput.HostIds[0], nil
 }
 
+func (e *AwsEC2) isHostAvailable(hostId *string) bool {
+	describeOutput, err := e.client.DescribeHosts(&ec2.DescribeHostsInput{
+		HostIds: aws.StringSlice([]string{aws.StringValue(hostId)}),
+	})
+	if err != nil {
+		klog.Errorf("cannot describe host %s: %v", *hostId, err)
+		return false
+	}
+	if len(describeOutput.Hosts) != 1 {
+		klog.Errorf("no such host: %s", aws.StringValue(hostId))
+		return false
+	}
+	host := describeOutput.Hosts[0]
+	return aws.StringValue(host.State) == AwsInstanceAvailableState
+}
+
+func (e *AwsEC2) waitForHostAvailable(ctx context.Context, hostId *string) bool {
```

Comment:
This function needs a timeout. Otherwise it may wait for the host to come up forever.

Reply:
> This function needs a timeout. Otherwise it may wait for the host to come up forever.

The caller specifies a timeout via `ctx`.